meta-balena-common/recipes-support/balena-hostname/balena-hostname/balena-hostname.service

```diff
@@ -2,10 +2,10 @@
 Description=Balena Hostname Configuration
 Requires=resin-boot.service balena-device-uuid.service bind-etc-hostname.service
 After=resin-boot.service balena-device-uuid.service bind-etc-hostname.service
+PartOf=config-json.target
 
 [Service]
 Type=oneshot
-RemainAfterExit=yes
```

Comment:
So there's a change of behaviour here that when avahi-daemon (re)starts it will re-run balena-hostname even if it has been run once already. Not necessarily a bad thing, just something to be aware of.

Reply:
@klutchell  That wasn't the intended behaviour. I thought that `RemainAfterExit` was only useful for services that had an `ExecStop` entry?
The observed behaviour is that a `config.json` change triggers `balena-hostname` and then `avahi` - which is what is required.

Reply:
Yeah, the services will both restart in the correct order as expected when config.json changes, I know you've tested that behaviour.

However, if avahi-daemon is restarted for any other reason (manually, failure, etc) it may restart the hostname.service it requires.
I think the same thing will happen with my recent changes to dnsmasq service, but I think it's fine and non-breaking.

1. change config.json manually
2. hostname.service restarts and stops
3. avahi-daemon restarts and stops
4. restart avahi-daemon manually
5. hostname.service is not running so it is (re)started

You can confirm that behaviour in case I'm mistaken, but it seems like an acceptable pattern.

Reply:
@klutchell Good spot. I think that it makes sense for me to reinstate the `RemainAfterExit=yes` for `balena-hostname` for the reasons you have described.

Your `dnsmasq` changes don't re-run `balena-net-config` when `dnsmasq` is restarted because `balena-net-config` has `RemainAfterExit=yes`. This seems much cleaner.